hook.php

```diff
@@ -252,6 +252,51 @@ function plugin_escalade_install() {
 
    $migration->migrationOneTable('glpi_plugin_escalade_configs');
 
+   if (!$DB->fieldExists('glpi_plugin_escalade_histories', 'previous_groups_id')
+      || ! !$DB->fieldExists("glpi_plugin_escalade_histories", 'counter')) {
+      $migration->addField('glpi_plugin_escalade_histories', 'previous_groups_id',
+                           'integer', ['before' => 'groups_id']);
+      $migration->addField('glpi_plugin_escalade_histories', 'counter',
+                           'integer', ['after' => 'groups_id']);
+      $migration->migrationOneTable('glpi_plugin_escalade_histories');
+
+      $history = new PluginEscaladeHistory();
+      $histories = [];
+      foreach ($history->find() as $data) {
+         $tickets_id = $data['tickets_id'];
+         unset($data['tickets_id']);
+
+         if (!isset($histories[$tickets_id])) {
+            $histories[$tickets_id] = [];
+         }
+
+         $histories[$tickets_id][] = $data;
+      }
+
+      foreach ($histories as $tickets_id => $h) {
+         echo 'Ticket #' . $tickets_id . '<br/>';
+         $counters = [];
+
+         foreach ($h as $k => $details) {
+            if (isset($h[$k+1])) {
+               $first  = $h[$k+1]['groups_id'] < $details['groups_id'] ? $h[$k+1]['groups_id'] : $details['groups_id'];
```

Comment:
I do not understand why there is such a numeric comparison on `groups_id` used to determines `$first` and `$second` values.
If `counter` is the number of escalade for one group, it should only be related to update from this group to another, right ?

Reply:
The counter field is not number of escalations for the current group (this could be done in simple query), but the number of escalations betwteen current group (A) and the previous one (B), the numeric comparaison is just one way to ensure the key (in array) used to store current count of escalations betwteen the 2 current groups is the same, whatever which group has escalated to other in the current row.

Reply:
OK, so it is only the counter for exact same escalation path if I understand right. Maybe the name "counter" is not enough precise in this case.

Reply:
yes it's the counter of total escalations for A > B or B > A